### DISCUSSION TITLE
Methods of writing USB sticks on various OSes

### DISCUSSION THREAD

**Discussion by wagnerck** [memtest86plus/memtest86plus#59] [General]
First, the new [memtest.org](https://memtest.org/) looks great. Very modern but not superfluous.

The "How to create a bootable Memtest86+ USB Drive using Linux?" and "How to create a bootable Memtest86+ USB Drive using MacOS" sections could use some info (especially the latter), and I have access to a whole bunch of different systems and tools, so I thought I'd contribute. I will very likely at some point be directing people to those downloads and instructions, and the easier I can make it for folks, the better.

Re: making the USB sticks under Linux, I know that the following tools work for building a stick from a memtest ISO file (either downloaded or built from source):
* The Disks app from Ubuntu and descendants has the option to restore IMG and ISO files to USB sticks, SD cards, etc. Select the device on the left, click on the three dots menu in the upper right and choose "Restore Disk Image".
* Pop!OS has a custom tool called Popsicle for writing USB sticks and the like from IMG and ISO files. It's available for other distros on [Flathub](https://flathub.org/apps/details/com.system76.Popsicle) and via [appimage](https://github.com/pop-os/popsicle/releases).
* [balenaEtcher](https://www.balena.io/etcher/) also has an appimage for Linux and it works well. It's also cross-platform so it may be worth treating it as a general suggestion for non-Windows systens?

As for MacOS, I've tested the following:
* Using the Restore function in Apple's Disk Utility does _not_ work, at least not in Monterey. It is unable to "validate source" so this may have something to do with what it expects an ISO file to contain.
* [balenaEtcher](https://www.balena.io/etcher/) works exactly as expected, same as it does under Linux.

Since there's already a dedicated installer for Windows, other options may not be necessary, but [Rufus](https://rufus.ie) is generally considered the winner for writing ISO files and the like on Windows right now, although balenaEtcher is significantly easier to use.

I'd love to hear what other tools folks are using.

**Comment by x86fr** [comment]
Thank you for your help,

I will add it to memtest.org (especially for MacOS). balenaEtcher seems to be a good solution for both Linux and MacOS.

About Windows, Rufus and balenaEtcher (or even win32diskimager) are all good solutions, but burning the ISO directly on an USB key has a serious drawback for (usually) less-experienced Windows users: they end up with a 4 MB Flash Key and they often don't know how to reclaim the "hidden" free space. I think Rufus can probably deal with this, but it's more work than just writing the ISO. Anyway, I will try to contact them (having Memtest86+ in the default list with FreeDOS would be awesome)
